harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -148,7 +148,35 @@ def make_workloads() -> workload.Stream:
         )
         controller.run()
 
-    def test_checkpointing_and_restoring(self, tmp_path: pathlib.Path) -> None:
+    @pytest.mark.gpu
+    @pytest.mark.parametrize(
+        "trial_class",
+        [
+            pytorch_xor_model.XORTrialWithLRScheduler,
+            pytorch_onevar_model.OneVarApexAMPTrial,
+            pytorch_onevar_model.OneVarAutoAMPTrial,
+            pytorch_onevar_model.OneVarManualAMPTrial,
+        ],
+        ids=[
+            "default",
+            "apex",
+            "autocast",
+            "manual",
+        ],
+    )
+    def test_checkpointing_and_restoring(self, trial_class, tmp_path: pathlib.Path) -> None:
```

Comment:
But this test is now very hard to read.  There are just too many if statements.  To me, that's a sign that the code paths are not actually similar enough to be combined.  Let's leave the original test alone and stick with a separate AMP test.

Reply:
I agree 100%. I was able to tell you my same feelings!